### DISCUSSION TITLE
Removal of few packages

### DISCUSSION THREAD

**Discussion by rkoystart** [python-poetry/poetry#7354] [Q&A]
The poetry version currently i am using is 1.1.6 and python version is 3.8.3.
I have multiple projects and each project has a pyproject.toml and a corresponding lock file.

During deployment purpose in few cases, i want to install a couple of projects in the same environment. But by doing so i could see that few packages are being removed.
For example say i have project1 and project 2 and its corresponding `pyproject1.toml` and `pyproject2.toml`. I create a new conda env and run the command `poetry install` which install all the dependencies in `pyproject1.toml` and then now again run the command `poetry install` which install `pyproject2.toml`, but removes certain packages which is necessary for the project1.

Can i know what is the reason for removal of those packages and is this the expected behaviour.
If yes is there any solution to this kind of problem.

**Comment by radoering** [comment]
Poetry 1.1.6 is quite old and not supported anymore. There have been lots of bug fixes in the meantime. Can you reproduce your issue with a current version of poetry? In current versions packages should be only removed when you call `poetry install --sync`. Without `--sync` packages are kept even if they are not required by the current project.
